socorro/processor/signature_utilities.py

```diff
@@ -462,42 +434,42 @@ def _create_frame_list(
         return frame_signatures_list
 
     def _get_crashing_thread(self, processed_crash):
-        return processed_crash.json_dump['crash_info']['crashing_thread']
+        return processed_crash['json_dump']['crash_info']['crashing_thread']
 
     def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-        if 'JavaStackTrace' in raw_crash and raw_crash.JavaStackTrace:
-            # generate a Java signature
+        # If this is a Java crash, then generate a Java signature
+        if 'JavaStackTrace' in raw_crash and raw_crash['JavaStackTrace']:
             signature, signature_notes = self.java_signature_tool.generate(
-                raw_crash.JavaStackTrace,
+                raw_crash['JavaStackTrace'],
                 delimiter=': '
             )
-            processed_crash.signature = signature
+            processed_crash['signature'] = signature
             if signature_notes:
-                processor_meta.processor_notes.extend(signature_notes)
+                processor_meta['processor_notes'].extend(signature_notes)
             return True
 
+        # This isn't a Java crash, so figure out what we need and then generate a C signature
         try:
             crashed_thread = self._get_crashing_thread(processed_crash)
         except KeyError:
             crashed_thread = None
+
         try:
             if processed_crash.get('hang_type', None) == 1:
-                # force the signature to come from thread 0
+                # Force the signature to come from thread 0
                 signature_list = self._create_frame_list(
-                    processed_crash.json_dump["threads"][0],
-                    processed_crash.json_dump['system_info']['os'] in
-                    "Windows NT"
+                    processed_crash['json_dump']['threads'][0],
+                    processed_crash['json_dump']['system_info']['os'] == 'Windows NT'
```

Comment:
I'm pretty sure this was a bug. Originally, it was checking to see if `processed_crash.json_dump['system_info']['os']` value was in the sequence `Windows NT`. That's probably not what we want. Maybe it was a list of multiple items before? Maybe the author meant `is` (though, that's wrong, too)`?

I changed it to `== 'Windows NT'`.

Anyhow, so before it would always pass in `False` to `make_modules_lower_case`. Now it might pass `True` if it's `Windows NT`. We shouldn't see that often, so I think this change even if it's wrong now won't have much effect on things.